### PR TITLE
BriefcaseConnection.activeSettings

### PR DESCRIPTION
**Pull request by pmconne**
Previously ToolAdmin.activeSettings specified the default model and category for use by placement tools, but those Ids are specific to an iModel.
Conceivably apps could contain multiple viewports in which editing tools can be used, but more realistically leftover Ids might remain when switching between iModels producing surprising results.
Moved the settings to BriefcaseConnection. Added events to notify when they change.
Note these settings are only defaults; more specialized tools will have their own logic for determining category and model.

### CODE REVIEW

**Comment by pmconne** [comment]
@bsteinbk I will need your assistance with ActiveSettingsWidget and ActiveSettingsManager.

**Comment by kabentley** [comment]
The term "settings" is confusing, since we use it for the backend concept "Setting". Probably there's nothing better for this context (I can't think of it), but this is the last chance to come up with something better.

**Comment by pmconne** [comment]
> The term "settings" is confusing, since we use it for the backend concept "Setting". Probably there's nothing better for this context (I can't think of it), but this is the last chance to come up with something better.

I had considered "active tool settings" but "tool settings" is already thing.
I had also considered "active placement settings" but I don't know that these settings will remain strictly about placement tools.

File: core/frontend/src/BriefcaseConnection.ts [hunk 1]
```diff
@@ -146,13 +146,68 @@ class ModelChangeMonitor {
   }
 }
 
+/** Default settings that can be used to control the behavior of [[Tool]]s that modify a [[BriefcaseConnection]].
```
Comment (kabentley) [inline core/frontend/src/BriefcaseConnection.ts:149]: i'd remove the word "Default". It implies there are other settings somewhere that override. You say they can be ignored.

**Comment by pmconne** [comment]
@kabentley `editorToolSettings`?

**Review by kabentley — APPROVED** [review]

**Comment by kabentley** [comment]
> editorToolSettings

I like that better.

File: core/frontend/src/BriefcaseConnection.ts (hunk 1 above)
Comment (pmconne) [reply core/frontend/src/BriefcaseConnection.ts]: Done; also renamed `BriefcaseConnection.activeSettings` to `editorToolSettings.

**Review by bbastings — APPROVED** [review]

**Review by mgooding — APPROVED** [review]

**Comment by bsteinbk** [comment]
> @pmconne - do you still need something from me?  I am not familiar with this code,  I believe Sam or Brien added it when initially working on an "editing" stage, but I am willing to make required changes.

**Review by bsteinbk — APPROVED** [review]
I verified that the Active Settings widget works to control the model and category of newly created elements.

**Pull request merged**